source/_posts/en/practice/can-we-use-dom.md

```diff
@@ -1,78 +1,76 @@
 ---
-title: 我们可以操作 DOM 吗？
+title: can use operate dom?
 categories:
 - practice
 ---
 
-
-我们在使用 San 的时候，特别是刚刚使用不久的新人，且 MVVM 框架的经验不是那么丰富，我们还是更习惯于使用 jQuery 作为类库来操作页面的交互，于是很自然的写出了这样的代码。
+When we were using San, especially the newcomers who just used it, and the experience of the MVVM framework is not so rich, are still more accustomed to using jQuery as a class library to manipulate the interaction of the page, so it is natural to write such code.
 
 ```
 var MyApp = san.defineComponent({
-    template: '<input value="没点" class="ipt"/><button class="btn"></button>',
+    template: '<input value="noClick" class="ipt"/><button class="btn"></button>',
     attached: function () {
         this.bindSomeEvents();
     },
     bindSomeEvents: function () {
         $('.btn').click(()=>{
-            $('.ipt').val('点了');
+            $('.ipt').val('clicked');
         });
     }
 });
 var myApp = new MyApp();
 myApp.attach(document.querySelector('#app'));
 ```
 
-然后用浏览器运行了这段程序，结果完全符合预期，完美~
+Then run this program in the browser, the results are completely in line with expectations, perfect.
 
-然而当我们进一步熟悉了 San 的使用方式后，对于上面的功能我们会写出这样的代码。
+However, as we become more familiar with the way San is used, we will write such code for the above functions.
 
 ```
 var MyApp = san.defineComponent({
-    template: '<div><input value="{{value}}"/><button on-click="clickHandler">点我</button></div>',
+    template: '<div><input value="{{value}}"/><button on-click="clickHandler">click me</button></div>',
     initData: function () {
         return {
-            value: '没点'
+            value: 'no click'
         };
     },
     clickHandler: function () {
-        this.data.set('value', '点了')
+        this.data.set('value', 'clicked')
     }
 });
 var myApp = new MyApp();
 myApp.attach(document.querySelector('#app'));
 ```
 
-仔细推敲了下这两段代码，不禁产生了一个疑问。
-
-直观的来看，San 的代码中我们直接调用 this.data.set 来修改某个属性的值，它自动将修改后的内容渲染到了 DOM 上，似乎看起来非常的神奇，但是它的根本上还是对 DOM 进行的操作，只不过这个操作是San框架帮你完成的，既然是这样，那我们为什么不能直接像第一段代码一样，直接修改，而要把这些操作交给 San 来完成呢？如果从性能上考虑交给 San 来做，它要完成从 Model 到视图上的关系绑定，还需要有一部分性能的损失，这样看起来代价还挺大的，那我们为什么还要这么做呢？
+Careful scrutiny of the two pieces of code, can not help but raise a question.
```

Comment:
One question may be raised up after we carefully compared these two pieces of code